src/shared_modules/keystore/CMakeLists.txt

```diff
@@ -14,12 +15,23 @@ if (NOT SRC_FOLDER)
     get_filename_component(SRC_FOLDER     ${CMAKE_SOURCE_DIR}/../../ ABSOLUTE)
 endif()
 
-include_directories(include/)
-include_directories(../utils/)
+if (NOT SHARED_MODULES)
+    get_filename_component(SHARED_MODULES ${CMAKE_SOURCE_DIR}/../ ABSOLUTE)
+endif()
+
+include_directories(include)
+
 include_directories(${SRC_FOLDER}/external/rocksdb/include)
 
-add_library(keystore STATIC
-    ${KEYSTORE_SRC}
-    )
+include_directories(${SHARED_MODULES}/utils)
+include_directories(${SHARED_MODULES}/common)
+
+link_directories(${SRC_FOLDER})
+
+add_library(keystore STATIC ${KEYSTORE_SRC})
+
+target_link_libraries(keystore rocksdb wazuhext)
+
+add_executable(wazuh-keystore main.cpp)
 
-target_link_libraries(keystore rocksdb)
+target_link_libraries(wazuh-keystore keystore)
```

Comment:
`asan` library should be added for `DEBUG` compilation, to avoid:
```shell
$  ./wazuh-keystore
==53513==ASan runtime does not come first in initial library list; you should either link runtime to your application or manually preload it with LD_PRELOAD.
``` 

you can see the use of this in:
https://github.com/wazuh/wazuh/blob/cdc44097c91876950185faddbff7e6902a0ee5e1/src/syscheckd/CMakeLists.txt#L111-L123

Reply:
Will be added in the corresponding Unit Tests issue

Reply:
This has nothing to do directly with the `UTs` but with the `DEBUG` compilation